platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/gossip/shadowgraph/ShadowgraphSynchronizer.java

```diff
@@ -141,37 +124,55 @@ public class ShadowgraphSynchronizer {
      */
     private final Duration nonAncestorFilterThreshold;
 
+    /**
+     * The current ancient mode.
+     */
+    private final AncientMode ancientMode;
+
+    /**
+     * Constructs a new ShadowgraphSynchronizer.
+     *
+     * @param platformContext        the platform context
+     * @param shadowGraph            stores events to sync
+     * @param numberOfNodes          number of nodes in the network
+     * @param syncMetrics            metrics for sync
+     * @param receivedEventHandler   events that are received are passed here
+     * @param fallenBehindManager    tracks if we have fallen behind
+     * @param intakeEventCounter     used for tracking events in the intake pipeline per peer
+     * @param executor               for executing read/write tasks in parallel
+     */
     public ShadowgraphSynchronizer(
             @NonNull final PlatformContext platformContext,
             @NonNull final Shadowgraph shadowGraph,
             final int numberOfNodes,
             @NonNull final SyncMetrics syncMetrics,
-            @NonNull final Supplier<GraphGenerations> generationsSupplier,
             @NonNull final Consumer<GossipEvent> receivedEventHandler,
             @NonNull final FallenBehindManager fallenBehindManager,
             @NonNull final IntakeEventCounter intakeEventCounter,
-            @NonNull final ParallelExecutor executor,
-            final boolean sendRecInitBytes,
-            @NonNull final InterruptableRunnable executePreFetchTips) {
+            @NonNull final ParallelExecutor executor) {
 
         Objects.requireNonNull(platformContext);
 
         this.time = platformContext.getTime();
         this.shadowGraph = Objects.requireNonNull(shadowGraph);
         this.numberOfNodes = numberOfNodes;
         this.syncMetrics = Objects.requireNonNull(syncMetrics);
-        this.generationsSupplier = Objects.requireNonNull(generationsSupplier);
         this.fallenBehindManager = Objects.requireNonNull(fallenBehindManager);
         this.intakeEventCounter = Objects.requireNonNull(intakeEventCounter);
         this.executor = Objects.requireNonNull(executor);
-        this.sendRecInitBytes = sendRecInitBytes;
-        this.executePreFetchTips = Objects.requireNonNull(executePreFetchTips);
         this.eventHandler = Objects.requireNonNull(receivedEventHandler);
 
         final SyncConfig syncConfig = platformContext.getConfiguration().getConfigData(SyncConfig.class);
         this.nonAncestorFilterThreshold = syncConfig.nonAncestorFilterThreshold();
 
         this.filterLikelyDuplicates = syncConfig.filterLikelyDuplicates();
+
+        this.ancientMode = platformContext
+                .getConfiguration()
+                .getConfigData(EventConfig.class)
+                .getAncientMode();
+
+        System.out.println("asdfasdf");
```

Comment:
asdfasdf.  - this is just here to see if I am paying attention, right?

Reply:
Lol, not the intention. But thanks for paying attention!

I was debugging a unit test, added this debug line ~4 hours into the endeavor. Let's just say it was "one of those problems". 🙃